modules/base/src/io/ivfvolumereader.cpp

```diff
@@ -56,8 +59,92 @@
 #include <type_traits>  // for remove_extent_t
 #include <memory_resource>
 
+#include <fmt/base.h>
+
 namespace inviwo {
 
+namespace {
+
+constexpr std::string_view InviwoVolume = "InviwoVolume";
+
+class Converter : public VersionConverter {
+public:
+    explicit Converter(int version);
+    virtual bool convert(TxElement* root) override;
+
+private:
+    void updateByteOrder(TxElement* root) const;
+
+    int version_;
+};
+
+Converter::Converter(int version) : version_(version) {}
+
+void Converter::updateByteOrder(TxElement* root) const {
+    auto metaDataMap = xml::getElement(root, "MetaDataMap");
+
+    if (metaDataMap) {
+        if (auto littleEndianNode =
+                xml::getElement(metaDataMap, "MetaDataItem&key=LittleEndian/MetaData")) {
+            // move byteorder to own tag
+            TxElement byteOrder{"ByteOrder"};
+
+            if (auto content = littleEndianNode->Attribute("content")) {
+                byteOrder.SetAttribute("content", content == "1" ? "LittleEndian" : "BigEndian");
+            } else {
+                byteOrder.SetAttribute("content", "LittleEndian");
+            }
+            root->InsertBeforeChild(metaDataMap, byteOrder);
+            metaDataMap->RemoveChild(littleEndianNode->Parent());
+        }
+    }
+}
+
+bool Converter::convert(TxElement* root) {
+    if (root->Value() != InviwoVolume) {
+        // ignore version number if the root node is not already an InviwoVolume, but for example an
+        // InviwoWorkspace
```

Comment:
```suggestion
```

Reply:
no that is correct

Reply:
@ResearchDaniel 

The reasoning here is that `InviwoWorkspace` has a different version number (3 at the moment) than the `InviwoVolume`. And for backwards compatibility InviwoWorkspace is still accepted for volumes and a warning is printed by the deserializer. However, in this case we cannot rely on the wrong version number in the converter.